src/graphql/admin/root/query/lightning-invoice.ts

```diff
@@ -11,7 +13,19 @@ const LightningInvoiceQuery = GT.Field({
   resolve: async (_, { hash }) => {
     if (hash instanceof Error) throw hash
 
-    const lightningInvoice = await lookupInvoiceByHash(hash)
+    const invoicesRepo = WalletInvoicesRepository()
+    const walletInvoice = await invoicesRepo.findByPaymentHash(hash)
```

Comment:
we cant do this, we are cleaning userinvoices and the purpose of this query (this is only a support query) is to look for a hash directly on lnd, should not be related to business logic.

Reply:
> Edit: graphql query should call only use cases in app layer, this should be moved to a new use case (maybe inside wallet)

yup ok

Reply:
> we cant do this, we are cleaning userinvoices (cron) and the purpose of this query (this is only a support query) is to look for a hash directly on lnd, should not be related to business logic.

we will also delete payment on lnd after this has been merged: https://github.com/GaloyMoney/galoy/pull/665

so really we should look into lnd and medici. but looping over lnd to find for a hash is not the best way to move forward. we know which lnd the hash belongs to, and the invoiceUser is currently only deleted after the invoice has expired ( > 90 days ).